File: experiments/016-LanguageComparisonRepeated/cells/java-gradle_1-Empty_free/2022-1c-recuperatorio-parcial1/20260917-053530-35565/analysis/own-tests-project/src/main/java/customerimporter/DNI.java
package customerimporter;

import java.util.function.Supplier;

public class DNI extends Identification {

    public static final String TYPE = "D";

    // error messages

    public static String invalidDNINumberErrorDescription() {
        return "Invalid DNI number";
    }

    // initialization

    public DNI(String aNumber) {
        super(aNumber);
    }

    protected void assertValidNumber(String aNumber) {
        assertValid(isAllDigits(aNumber));
        int dniNumber = Integer.parseInt(aNumber);
        assertValid(dniNumber >= 1 && dniNumber <= 99999999);
    }

    protected String invalidValueErrorDescription() {
        return invalidDNINumberErrorDescription();
    }

    // accessing

    public String type() {
        return TYPE;
    }

    // testing

    public boolean isDNI() {
        return true;
    }

    public boolean isCUIT() {
        return false;
    }

    public Object dniNumberIfNone(Supplier<Object> aNoneBlock) {
        return Integer.parseInt(number());
    }

    public Object cuitNumberIfNone(Supplier<Object> aNoneBlock) {
        return aNoneBlock.get();
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-202259-53403/analysis/own-tests-project/src/customerimporter/dni.py
from customerimporter.identification import Identification


class Dni(Identification):

    # type

    @classmethod
    def type_code(cls):
        return "D"

    # error messages

    @classmethod
    def invalid_value_error_description(cls):
        return "Invalid DNI number"

    # validation

    @classmethod
    def assert_is_valid(cls, a_number):
        cls.assert_all_are_digits(a_number)
        cls.assert_is_between(int(a_number), 1, 99999999)

    # testing

    def is_dni(self):
        return True

    # number

    def dni_number_if_none(self, a_none_block):
        return int(self.number())
